### DISCUSSION TITLE
When using NLOHMANN_DEFINE_TYPE_INTRUSIVE, Stop the exception if dont't have the key-value pairs

### DISCUSSION THREAD

**Discussion by i-curve** [nlohmann/json#4235] [Ideas]
Hi, I like this library very much and I use it a lot.

When I use the NLOHMANN_DEFINE_TYPE_INTRUSIVE macro, I encountered some differences in usage habits.
the ```NLOHMANN_DEFINE_TYPE_INTRUSIVE``` implements ```to_json``` and ```from_json``` functions.
The detail of ```from_json``` will use ```NLOHMANN_JSON_FROM```
the defination of ```NLOHMANN_JSON_FROM``` is
```c++
#define NLOHMANN_JSON_FROM(v1) nlohmann_json_j.at(#v1).get_to(nlohmann_json_t.v1);
```

if the json string don't have the key, it will throw an exception and terminal the program.
In my use case, if the key-value pair doesn't exist, you should ignore it instead of terminating the program.so I think it would be better to ignore the key-value pairs:

```c++
#define NLOHMANN_JSON_FROM(v1) if(nlohmann_json_j.contains(#v1)) nlohmann_json_j.at(#v1).get_to(nlohmann_json_t.v1);
```

@nlohmann Which method is better?

**Comment by gregmarr** [comment]
See `NLOHMANN_DEFINE_TYPE_INTRUSIVE_WITH_DEFAULT`

**Comment by i-curve** [comment]
I never knew about this method and didn't see it in the readme.
hope add the ```NLOHMANN_DEFINE_TYPE_INTRUSIVE_WITH_DEFAULT``` and ```.value()``` method in the readme

**Reply by gregmarr** [reply]
Now #4260 for the macros.
